packages/atomic/src/components/atomic-search-box/atomic-search-box.tsx

```diff
@@ -112,18 +117,30 @@ export class AtomicSearchBox implements AtomicSearchBoxOptions {
   }
 
   private onClickSuggestion(e: MouseEvent) {
-    const value = (e.target as HTMLButtonElement).value;
-    this.searchBox.selectSuggestion(value);
+    const value = (e.currentTarget as HTMLLIElement).value;
+    this.searchBox.selectSuggestion(
+      this.searchBoxState.suggestions[value].rawValue
+    );
   }
 
   private get submitButton() {
     return (
       <button
         type="button"
         part="submit-button"
+        class={
+          'w-10 bg-transparent border-0 focus:outline-none border-medium-grey border-solid p-0 ' +
+          (this.leadingSubmitButton ? 'border-r' : 'border-l')
+        }
+        aria-label={this.context.i18n.t('search')}
         onClick={() => this.searchBox.submit()}
       >
-        {this.context.i18n.t('search')}
+        <slot name="submit-button">
+          <div
+            innerHTML={SearchIcon}
```

Comment:
The `<SearchIcon/>` style syntax if that's what you're referring too isn't actually naively supported by Stencil, since it doesn't properly expand the `<SearchIcon/>` into the the html actually contained inside `SearchIcon.svg`. We already had [Stencil Inline SVG](https://github.com/fabriciomendonca/stencil-inline-svg) installed, and this is the syntax it supports so I used that.